bgpd/bgp_route.c

```diff
@@ -8161,28 +8160,26 @@ DEFUN (show_ip_bgp_ipv4,
        SHOW_STR
        IP_STR
        BGP_STR
-       "Address family\n"
-       "Address Family modifier\n"
+       "Address family\n" 
+       "Address Family modifier\n" 
        "Address Family modifier\n"
        "JavaScript Object Notation\n")
 {
   u_char uj = use_json(argc, argv);
-
-  if (strncmp (argv[0], "m", 1) == 0)
-    return bgp_show (vty, NULL, AFI_IP, SAFI_MULTICAST, bgp_show_type_normal,
-                     NULL, uj);
  
-  return bgp_show (vty, NULL, AFI_IP, SAFI_UNICAST, bgp_show_type_normal, NULL, uj);
+  return bgp_show (vty, NULL, AFI_IP, 
+                   bgp_vty_safi_from_arg(argv[0]), 
+                   bgp_show_type_normal, NULL, uj);
 }
 
 ALIAS (show_ip_bgp_ipv4,
        show_bgp_ipv4_safi_cmd,
        "show bgp ipv4 (unicast|multicast) {json}",
        SHOW_STR
        BGP_STR
-       "Address family\n"
-       "Address Family modifier\n"
-       "Address Family modifier\n"
+       "Address family\n" 
```

Comment:
@donaldsharp WRT && (mpath_cfg && CHECK_FLAG(....) - no as this is a passed parameter which may be NULL, i.e. not provide any config info.